### ISSUE TITLE
Capistrano deploy broken

### ISSUE THREAD

**Issue by bryanleetc** [rails/webpacker#533]
I managed to make a couple of successful deploys initially. After making a few changes to my Rails code (webpacker related code and javascript all untouched), deploys are suddenly broken. The deploy fails when Webpacker is compiling assets. I only have a very cryptic error message:

```
yarn run v0.24.6

$ "/home/root/apps/aixi/releases/20170624123842/node_modules/.bin/webpack" --config /home/root/apps/aixi/releases/20170624123842/config/webpack/production.js

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.



(node:30203) DeprecationWarning: Chunk.modules is deprecated. Use Chunk.getNumberOfModules/mapModules/forEachModule/containsModule instead.

Killed

 DEBUG [379edb7f]

error Command failed with exit code 137.
```

Running the last command (below) in the server directly finishes successfully:
```
"/home/root/apps/aixi/releases/20170624123842/node_modules/.bin/webpack" --config /home/root/apps/aixi/releases/20170624123842/config/webpack/production.js
```

**Comment by ytbryan** [comment]
Perhaps share more info on what `After making a few changes to my Rails code `

**Comment by bryanleetc** [comment]
Removing ActionCable and enabling encrypted secrets. I've tried disabling encrypted secrets again but the compile is still failing.

With encrypted secrets disabled, running `NODE_ENV=production "/home/root/apps/aixi/releases/20170624123842/node_modules/.bin/webpack" --config /home/root/apps/aixi/releases/20170624123842/config/webpack/production.js` in my production server yields:

```
ERROR in ./node_modules/extract-text-webpack-plugin/loader.js?{"omit":1,"remove":true}!./node_modules/vue-style-loader!./node_modules/css-loader?minimize!./node_modules/vue-loader/lib/style-compiler?{"vue":true,"id":"data-v-29083192","scoped":true,"hasInlineConfig":false}!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./app/javascript/packs/app.vue
Module build failed: Error: "extract-text-webpack-plugin" loader is used without the corresponding plugin, refer to https://github.com/webpack/extract-text-webpack-plugin for the usage example
    at Object.module.exports.pitch (/home/root/apps/aixi/releases/20170624141900/node_modules/extract-text-webpack-plugin/loader.js:25:9)
 @ ./app/javascript/packs/app.vue 2:2-358
```

Webpack config has been untouched so far

**Comment by gauravtiwari** [comment]
@bryanleetc You got `extract-text-webpack-plugin` initialised in your shared.js right?

**Comment by bryanleetc** [comment]
I edited my comment above. The `extract-text-webpack-plugin` related error was triggered as I was running the webpack command using an outdated capistrano release directory. Running the code using the `current` directory solves it.

The problem still exists when trying to deploy with Capistrano.

**Comment by gauravtiwari** [comment]
@bryanleetc Yes, I see - seems related to #531. I will push up something to fix this 👍

**Comment by bryanleetc** [comment]
I overwrote the `webpacker:compile` rake task to output the `env` local variable during compile, outputting:
```
`{ "NODE_ENV" => 'production', "ASSET_HOST" => nil }
```

If I run ./bin/webpack directly: `NODE_ENV=production ASSET_HOST=nil ./bin/webpack`, it falis.

However, `NODE_ENV=production ./bin/webpack` passes in my production server.

Interestingly, if I overwrote env variable in the webpacker:compile rake task to:  `env = { "NODE_ENV" => Webpacker.env }.freeze`, the deploy still fails.

Hope this can help!

**Comment by bryanleetc** [comment]
Also, if it helps, in the production server, running `RAILS_ENV=production bundle exec rails webpacker:compile` works, but `RAILS_ENV=production bundle exec rails assets:precompile` fails.

**Comment by bryanleetc** [comment]
I edited bin/webpack to make it output verbose and the results of the failed deploy as follows:

```
[Webpacker] Compilation Failed
yarn run v0.24.6
verbose 0.57 Checking for configuration file "/home/root/apps/aixi/releases/20170624160416/.npmrc".
verbose 0.573 Checking for configuration file "/usr/local/share/.npmrc".
verbose 0.574 Checking for configuration file "/usr/.npmrc".
verbose 0.575 Checking for configuration file "/home/root/apps/aixi/releases/20170624160416/.npmrc".
verbose 0.575 Checking for configuration file "/home/root/apps/aixi/releases/.npmrc".
verbose 0.575 Checking for configuration file "/home/root/apps/aixi/.npmrc".
verbose 0.576 Checking for configuration file "/home/root/apps/.npmrc".
verbose 0.576 Checking for configuration file "/home/root/.npmrc".
verbose 0.577 Checking for configuration file "/home/.npmrc".
verbose 0.579 Checking for configuration file "/home/root/apps/aixi/releases/20170624160416/.yarnrc".
verbose 0.58 Checking for configuration file "/usr/local/share/.yarnrc".
verbose 0.581 Found configuration file "/usr/local/share/.yarnrc".
verbose 0.585 Checking for configuration file "/usr/.yarnrc".
verbose 0.585 Checking for configuration file "/home/root/apps/aixi/releases/20170624160416/.yarnrc".
verbose 0.586 Checking for configuration file "/home/root/apps/aixi/releases/.yarnrc".
verbose 0.586 Checking for configuration file "/home/root/apps/aixi/.yarnrc".
verbose 0.587 Checking for configuration file "/home/root/apps/.yarnrc".
verbose 0.587 Checking for configuration file "/home/root/.yarnrc".
verbose 0.587 Checking for configuration file "/home/.yarnrc".
verbose 0.6 current time: 2017-06-24T16:05:40.105Z
$ "/home/root/apps/aixi/releases/20170624160416/node_modules/.bin/webpack" --config /home/root/apps/aixi/releases/20170624160416/config/webpack/production.js
verbose 8.604 Error: Command failed with exit code 135.
    at MessageError (/usr/share/yarn/lib/yarn-cli.js:138:5)
    at /usr/share/yarn/lib/yarn-cli.js:37362:15
    at throw (native)
    at step (/usr/share/yarn/lib/yarn-cli.js:91:30)
    at /usr/share/yarn/lib/yarn-cli.js:104:13
    at process._tickCallback (internal/process/next_tick.js:109:7)
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Bus error (core dumped)
error Command failed with exit code 135.

rake stderr: Nothing written


** DEPLOY FAILED
** Refer to log/capistrano.log for details. Here are the last 20 lines:


verbose 0.585 Checking for configuration file "/home/root/apps/aixi/releases/20170624160416/.yarnrc".

verbose 0.586 Checking for configuration file "/home/root/apps/aixi/releases/.yarnrc".

verbose 0.586 Checking for configuration file "/home/root/apps/aixi/.yarnrc".

verbose 0.587 Checking for configuration file "/home/root/apps/.yarnrc".

verbose 0.587 Checking for configuration file "/home/root/.yarnrc".

verbose 0.587 Checking for configuration file "/home/.yarnrc".

verbose 0.6 current time: 2017-06-24T16:05:40.105Z

$ "/home/root/apps/aixi/releases/20170624160416/node_modules/.bin/webpack" --config /home/root/apps/aixi/releases/20170624160416/config/webpack/production.js

verbose 8.604 Error: Command failed with exit code 135.

    at MessageError (/usr/share/yarn/lib/yarn-cli.js:138:5)

    at /usr/share/yarn/lib/yarn-cli.js:37362:15

    at throw (native)

    at step (/usr/share/yarn/lib/yarn-cli.js:91:30)

    at /usr/share/yarn/lib/yarn-cli.js:104:13

    at process._tickCallback (internal/process/next_tick.js:109:7)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.



 DEBUG [11b551cb] 	Bus error (core dumped)

error Command failed with exit code 135.
```

**Comment by bryanleetc** [comment]
Solved it 😅
Exit code 135 was solved by adding `npm rebuild node-sass` to the postinstall hooks.

After that, exit code 137 started happening, which I traced to the most silly reason of the server not having enough memory. This is a new project so I've been testing it on a very small VPS (512mb ram) and it never occured to me that it was because of a lack of memory. Once I added in some swap, the issue was resolved.
